### PR TITLE
Fix exclusion directory and file matching and add support for reading a top level gitignore file

### PR DESCRIPTION
**Pull request by rabbah**
Closes https://github.com/apache/incubator-openwhisk-utilities/issues/39.
This patch bundles source code from `pathspec` https://pypi.org/project/pathspec/ and updates the LICENSE accordingly. This is to avoid a `pip install` in all downstream clients.

The pathspec library is Mozilla Public License 2.0 [1]. My reading of the Mozilla License, their FAQ [2] and the Apache 3rd Party License Policy [3] leads me to conclude the bundling is acceptable. 

[1] https://www.mozilla.org/en-US/MPL/2.0/
[2] https://www.mozilla.org/en-US/MPL/2.0/FAQ
[3] https://apache.org/legal/resolved.html

### CODE REVIEW

**Comment by dgrove-oss** [comment]
@rabbah -- are we ready to merge this or do we need fixups in some of the scanned repos to fix problems in files that we had previously been erroneously ignoring?

**Comment by rabbah** [comment]
Ready to merge as long as someone has sanity checked my understanding of the licenses and the inclusion of the library. The patch will not require downstream changes as is.

File: scancode/scanCode.py [hunk 1]
```diff
@@ -549,7 +554,7 @@ def is_dir(path):
             print_warning(WARN_SCAN_EXCLUDED_PATH % excluded_path)
 
     # Display which files where excluded from these paths
-    if args.display_exclusions:
+    if VERBOSE:
```
Comment (mrutkows) [inline scancode/scanCode.py]: Sometimes the listing of exclusions is not desirable unto itself (even with VERBOSE) enabled; would prefer leaving this as a sep. option in addition to general verbose.

File: licenses/LICENSE-pathspec.txt [hunk 2]
```diff
@@ -0,0 +1,374 @@
+Mozilla Public License Version 2.0
```
Comment (mrutkows) [inline licenses/LICENSE-pathspec.txt]: TYVM for adding this and ref. the source license diligently.

File: scancode/scanCode.py (hunk 1 above)
Comment (rabbah) [reply scancode/scanCode.py]: Reverted.

**Comment by rabbah** [comment]
Thanks @mrutkows and @dgrove-oss for the review.

**Pull request merged**